assets/queries/ansible/aws/port22_in_public_scope/query.rego

```diff
@@ -0,0 +1,81 @@
+package Cx
+
+CxPolicy[result] {
+    document := input.document[i]
+    tasks := getTasks(document)
+    task := tasks[t]
+    
+    currentFromPort:= task["amazon.aws.ec2_group"].rules[index].from_port
+    currentToPort := task["amazon.aws.ec2_group"].rules[index].to_port
+    
+    cidr := task["amazon.aws.ec2_group"].rules[index].cidr_ip
+    
+    isSSH(currentFromPort, currentToPort)
+    not isPrivate(cidr)
+
+    result := {
+        "documentId":        document.id,
+        "searchKey":         sprintf("name={{%s}}.{{amazon.aws.ec2_group}}.rules", [task.name]),
+        "issueType":         "IncorrectValue",
+        "keyExpectedValue":  sprintf("amazon.aws.ec2_group.rules[%d] SSH' (Port:22) is not public", [index]),
+        "keyActualValue": 	 sprintf("amazon.aws.ec2_group.rules[%d] SSH' (Port:22) is public", [index])
+    }
+}
+
+CxPolicy[result] {
+    document := input.document[i]
+    tasks := getTasks(document)
+    task := tasks[t]
+    
+    currentFromPort:= task["amazon.aws.ec2_group"].rules[index].from_port
+    currentToPort := task["amazon.aws.ec2_group"].rules[index].to_port
+    
+    cidr := task["amazon.aws.ec2_group"].rules[index].cidr_ipv6
+    
+    isSSH(currentFromPort, currentToPort)
+    not isPrivate(cidr)
+
+    result := {
+        "documentId":        document.id,
+        "searchKey":         sprintf("name={{%s}}.{{amazon.aws.ec2_group}}.rules", [task.name]),
+        "issueType":         "IncorrectValue",
+        "keyExpectedValue":  sprintf("amazon.aws.ec2_group.rules[%d] SSH' (Port:22) is not public", [index]),
+        "keyActualValue": 	 sprintf("amazon.aws.ec2_group.rules[%d] SSH' (Port:22) is public", [index])
+    }
+}
+
+getTasks(document) = result {
+    result := [body | playbook := document.playbooks[0]; body := playbook.tasks]
+    count(result) != 0
+} else = result {
+    result := [body | playbook := document.playbooks[_]; body := playbook ]
+    count(result) != 0
+}
+
+isSSH(currentFromPort, currentToPort) = allow {
+     currentFromPort <= 22
+     currentToPort >= 22
+     allow = true
+}
```

Comment:
If the 'from' or 'to' ports are -1 means that all ports are considered, you should check those cases as well

Reply:
also, if you do not indicate the result from the function, i.e, `= allow`, by default returns boolean and so returns true if all the statements inside are true, so you don't need to specify the return if its boolean